Remotion/ObjectBinding/Web.UnitTests/UI/Controls/Validation/CompoundValidationFailureMatcherTest.cs

```diff
@@ -0,0 +1,135 @@
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using Moq;
+using NUnit.Framework;
+using Remotion.ObjectBinding.BusinessObjectPropertyPaths.Results;
+using Remotion.ObjectBinding.Validation;
+using Remotion.ObjectBinding.Web.UI.Controls.Validation;
+
+namespace Remotion.ObjectBinding.Web.UnitTests.UI.Controls.Validation
+{
+  [TestFixture]
+  public class CompoundValidationFailureMatcherTest
+  {
+    [Test]
+    public void GetMatchingValidationFailures_WithEmptyPropertyPaths_ReturnsEmpty ()
+    {
+      var rowObject = new Mock<IBusinessObject>();
+      var validationResult = new Mock<IBusinessObjectValidationResult>();
+
+
+      var matcher = new CompoundValidationFailureMatcher(Array.Empty<IBusinessObjectPropertyPath>());
+      var result = matcher.GetMatchingValidationFailures(rowObject.Object, validationResult.Object);
+
+      Assert.That(result, Is.EqualTo(Array.Empty<BusinessObjectValidationFailure>()));
+    }
+
+    [Test]
+    public void GetMatchingValidationFailures_ReturnsValidationFailures ()
+    {
+      var rowObject = new Mock<IBusinessObject>();
+      var validationResultMock = new Mock<IBusinessObjectValidationResult>();
+
+      var firstPropertyPathResultMock = new Mock<IBusinessObjectPropertyPathResult>();
+      var firstResultObjectStub = new Mock<IBusinessObject>();
+      firstPropertyPathResultMock.Setup(_ => _.ResultObject).Returns(firstResultObjectStub.Object);
+      var firstResultPropertyStub = new Mock<IBusinessObjectProperty>();
+      firstPropertyPathResultMock.Setup(_ => _.ResultProperty).Returns(firstResultPropertyStub.Object);
+      var firstPropertyPathStub = new Mock<IBusinessObjectPropertyPath>();
+
+      firstPropertyPathStub.Setup(_ => _.GetResult(rowObject.Object, BusinessObjectPropertyPath.UnreachableValueBehavior.ReturnNullForUnreachableValue, BusinessObjectPropertyPath.ListValueBehavior.FailForListProperties))
+          .Returns(firstPropertyPathResultMock.Object).Verifiable();
+
+      var secondPropertyPathResultMock = new Mock<IBusinessObjectPropertyPathResult>();
+      var secondResultObjectStub = new Mock<IBusinessObject>();
+      secondPropertyPathResultMock.Setup(_ => _.ResultObject).Returns(secondResultObjectStub.Object);
+      var secondResultPropertyStub = new Mock<IBusinessObjectProperty>();
+      secondPropertyPathResultMock.Setup(_ => _.ResultProperty).Returns(secondResultPropertyStub.Object);
+
+      var secondPropertyPathStub = new Mock<IBusinessObjectPropertyPath>();
+      secondPropertyPathStub.Setup(_ => _.GetResult(rowObject.Object, BusinessObjectPropertyPath.UnreachableValueBehavior.ReturnNullForUnreachableValue, BusinessObjectPropertyPath.ListValueBehavior.FailForListProperties))
+          .Returns(secondPropertyPathResultMock.Object).Verifiable();
```

Comment:
```suggestion
      firstPropertyPathResultMock
          .Setup(_ => _.ResultObject)
          .Returns(firstResultObjectStub.Object);

      var firstResultPropertyStub = new Mock<IBusinessObjectProperty>();
      firstPropertyPathResultMock
          .Setup(_ => _.ResultProperty)
          .Returns(firstResultPropertyStub.Object);

      var firstPropertyPathStub = new Mock<IBusinessObjectPropertyPath>();
      firstPropertyPathStub
          .Setup(_ => _.GetResult(rowObject.Object, BusinessObjectPropertyPath.UnreachableValueBehavior.ReturnNullForUnreachableValue, BusinessObjectPropertyPath.ListValueBehavior.FailForListProperties))
          .Returns(firstPropertyPathResultMock.Object)
          .Verifiable();

      var secondPropertyPathResultMock = new Mock<IBusinessObjectPropertyPathResult>();
      var secondResultObjectStub = new Mock<IBusinessObject>();
      secondPropertyPathResultMock
          .Setup(_ => _.ResultObject)
          .Returns(secondResultObjectStub.Object);

      var secondResultPropertyStub = new Mock<IBusinessObjectProperty>();
      secondPropertyPathResultMock
          .Setup(_ => _.ResultProperty)
          .Returns(secondResultPropertyStub.Object);

      var secondPropertyPathStub = new Mock<IBusinessObjectPropertyPath>();
      secondPropertyPathStub
          .Setup(_ => _.GetResult(rowObject.Object, BusinessObjectPropertyPath.UnreachableValueBehavior.ReturnNullForUnreachableValue, BusinessObjectPropertyPath.ListValueBehavior.FailForListProperties))
          .Returns(secondPropertyPathResultMock.Object)
          .Verifiable();
```
I think adding line breaks here, makes it more readable?

Reply:
Yeah I agree. I still don't think that its particularly readable, as there are so many mocks and some more or less required verbosity.